src/event/model/event.rs

```diff
@@ -251,3 +253,84 @@ impl EventPayload {
         Ok(event)
     }
 }
+
+#[derive(Debug, Eq, PartialEq, Serialize)]
+pub struct Events {
+    events: Vec<Event>,
+}
+
+impl Events {
+    pub async fn fetch(
+        after: i32,
+        max_events: i32,
+        pool: &MySqlPool,
+    ) -> Result<Events, sqlx::Error> {
+        let event_ids = Events::fetch_event_ids(after, max_events, pool).await?;
+
+        let mut event_tasks = FuturesOrdered::new();
+        for event_id in event_ids {
+            event_tasks.push(Events::fetch_event(event_id, pool));
+        }
```

Comment:
That looks super inefficient. Doesn't this trigger `max_events` sql queries in sequence?

Reply:
Yep :see_no_evil: -> Let's see whether there is a bette impl